ir/instr.cpp

```diff
@@ -594,6 +598,24 @@ StateValue BinOp::toSMT(State &s) const {
     };
     break;
 
+  case FMinimum:
+  case FMaximum:
+    fn = [&](auto a, auto ap, auto b, auto bp) -> StateValue {
+      auto v = [&](expr &a, expr &b) {
+        expr zpos = expr::mkNumber("0", a), zneg = expr::mkNumber("-0", a);
+        expr cmp = (op == FMinimum) ? a.fole(b) : a.foge(b);
+        expr neg_cond = (op == FMinimum) ? a.isFPNeg() || b.isFPNeg()
+                                         : a.isFPNeg() && b.isFPNeg();
```

Comment:
is this bit needed? Did you check the spec to see it smt already does the right thing?  (it's very likely as LLVM and SMT aren't inventing different semantics)

Reply:
Yeah it is needed.  The SMT less than returns false if both operand is either +0 or -0 but LLVM's maximum and minimum are sensitive to the sign of zeros. 

See https://github.com/Z3Prover/z3/blob/72d140334fff2d285956b8419b552d727619156a/src/ast/fpa/fpa2bv_converter.cpp#L2237